### ISSUE TITLE
TFT model prediction is shifted on the y-axis

### ISSUE THREAD

**Issue by erl61** [unit8co/darts#1667]
Hello!
I am new to neural networks, so I would be glad for any advice!

I am trying to use a Temporal Fusion Transformer to predict a financial timeseries based on future covariates. As covariates I use datetime features such as 'month', 'dayofweek' etc. I have multiple datasets of different products with daily data for four years (~1500 points).
When I try to predict a target value using an example from documentation I get good predictions for some datasets. However, for the other datasets predictions appear to be shifted. The first predicted value significantly lower than the previous factual value, while trend and seasonality look good.

I have tried to increase the number of training epochs, hidden size, LSTM layers in different combinations, but it hasn't fixed the problem..

    series = TimeSeries.from_dataframe(my_df, time_col='ds', fill_missing_dates=True, freq='D')

    # Create training and validation sets:
    train, val = series.split_after(pd.Timestamp('2023-03-20'))

    # Normalize the time series (note: we avoid fitting the transformer on the validation set)
    # Change name
    scaler = Scaler()
    train_scaled = scaler.fit_transform(train)

    # create year, month and integer index covariate series
    covariates = datetime_attribute_timeseries(series, attribute="month", one_hot=False)

    covariates = covariates.stack(
        datetime_attribute_timeseries(series, attribute="weekofyear", one_hot=False)
    )
    covariates = covariates.stack(
        datetime_attribute_timeseries(series, attribute="day_of_year", one_hot=False)
    )
    covariates = covariates.stack(
        datetime_attribute_timeseries(series, attribute="day_of_week", one_hot=False)
    )
    covariates = covariates.stack(
        datetime_attribute_timeseries(series, attribute="quarter", one_hot=False)
    )

    covariates = covariates.stack(
        datetime_attribute_timeseries(series, attribute="is_month_start", one_hot=False)
    )

    covariates = covariates.stack(
        datetime_attribute_timeseries(series, attribute="is_month_end", one_hot=False)
    )

    covariates = covariates.stack(
        datetime_attribute_timeseries(series, attribute="is_quarter_start", one_hot=False)
    )

    covariates = covariates.stack(
        datetime_attribute_timeseries(series, attribute="is_quarter_end", one_hot=False)
    )

    covariates = covariates.stack(
        datetime_attribute_timeseries(series, attribute="is_year_start", one_hot=False)
    )

    covariates = covariates.stack(
        datetime_attribute_timeseries(series, attribute="is_year_end", one_hot=False)
    )

    covariates = covariates.stack(
        TimeSeries.from_times_and_values(
            times=series.time_index,
            values=np.arange(len(series)),
            columns=["linear_increase"],
        )
    )
    covariates = covariates.astype(np.float32)

    scaler_covs = Scaler()

    covariates_transformed = scaler_covs.fit_transform(covariates)
        # default quantiles for QuantileRegression
    quantiles = [
        0.01,
        0.05,
        0.1,
        0.15,
        0.2,
        0.25,
        0.3,
        0.4,
        0.5,
        0.6,
        0.7,
        0.75,
        0.8,
        0.85,
        0.9,
        0.95,
        0.99,
    ]
    input_chunk_length = 112
    forecast_horizon = 112
    my_model = TFTModel(
        input_chunk_length=input_chunk_length,
        output_chunk_length=forecast_horizon,
        hidden_size=16,
        lstm_layers=2,
        num_attention_heads=4,
        dropout=0.1,
        batch_size=32,
        n_epochs=100,
        add_relative_index=False,
        add_encoders=None,
        full_attention=True,
        likelihood=QuantileRegression(
            quantiles=quantiles)
    )

    my_model.fit(train_scaled, 
             future_covariates=covariates_transformed)

    my_model.predict(n=364, num_samples=50)

<img width="886" alt="pic" src="https://user-images.githubusercontent.com/18754682/227335561-c45dfab3-5f38-4a54-a83d-27ab1189ab8b.png">

When I manually shift predictions to the same level as previous fact it shows a good accuracy in terms of mape, so seasonality and trend is fine.

Thank you!

**Comment by madtoinou** [comment]
Hi,

It seems like the model is actually not properly capturing the trend since the predictions are shifted on the y-axis. It looks like the model believes that one of the sudden drop is occurring at the very beginning of the prediction period, instead of a couple of timestamps ago (visible in the last chunk of the blue curve).

In my opinion, there is not much you can do aside from trying other models architectures or increase your training set.
